<a href="https://colab.research.google.com/github/igjorque/IM_S6_DialogForAlert/blob/master/Entregable1_IgnacioJorquera.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import files
files.upload() #upload kaggle.json -> API key de kaggle

!pip install -q kaggle
!mkdir -p ~/.kaggle
!cp kaggle.json ~/.kaggle/
!ls ~/.kaggle
!chmod 600 /root/.kaggle/kaggle.json

Saving kaggle.json to kaggle.json
kaggle.json


In [4]:
!kaggle competitions download -c ia2021

  0% 0.00/328k [00:00<?, ?B/s]
100% 328k/328k [00:00<00:00, 45.1MB/s]
  0% 0.00/111k [00:00<?, ?B/s]
100% 111k/111k [00:00<00:00, 90.5MB/s]


In [1]:
import pandas as pd
testdf = pd.read_csv('test.csv')
traindf = pd.read_csv('train.csv')

---

Primero, visualizamos los datos de entrenamiento con los que trabajaremos.

In [30]:
traindf

SL      EEG  BP   HR  CIRCULATION  ACTIVITY
0        4019.64 -1600.00  13   79          317         3
1        2191.03 -1146.08  20   54          165         2
2        2787.99 -1263.38  46   67          224         2
3        9545.98 -2848.93  26  138          554         4
4       14148.80 -2381.15  85  120          809         4
...          ...      ...  ..  ...          ...       ...
11994   52550.30 -2970.00  31  188         2599         3
11995   26761.00 -2776.53  87  156         1521         4
11996   36737.00   555.00  60  234         1581         2
11997   39610.60 -3110.00  17  196         1885         3
11998  280014.00 -9684.44  68  468         9276         5

[11999 rows x 6 columns]

---

**Pregunta 1**

Muestra la distribución de valores del descriptor `Activity`. ¿Cuál de ellos tiene un mayor porcentaje sobre el total, y cuánto es? 

**Respuesta**

Tenemos el siguiente total de datos:

In [3]:
len(traindf)

11999

Y podemos ver que se distribuyen de la siguiente manera:

In [27]:
traindf['ACTIVITY'].value_counts()

0    3344
3    2625
4    2569
2    1849
5    1241
1     371
Name: ACTIVITY, dtype: int64

Vemos que la `Activity = 0` (De pie) es la que tiene el mayor número de ellos. Calculamos ahora el porcentaje que representa:

In [6]:
(max(traindf['ACTIVITY'].value_counts())/len(traindf))*100

27.868989082423536

También podemos obtener todas las proporciones de la siguiente forma:

In [26]:
traindf['ACTIVITY'].value_counts(normalize=True)

0    0.278690
3    0.218768
4    0.214101
2    0.154096
5    0.103425
1    0.030919
Name: ACTIVITY, dtype: float64

Por tanto, la `Activity = 0` es la que tiene un mayor porcentaje sobre el total, con un 27.87% (aprox.)

___

**Pregunta 2**

¿Cuál es la media del descriptor `Circulation` para los datos cuya `Activity` sea la de mayor porcentaje?

**Respuesta**

Sabemos que la `Activity = 0` es la que mayor porcentaje tiene sobre el total. Por tanto, calculamos la media de su descriptor `Circulation` de la siguiente forma:

In [8]:
traindf[traindf['ACTIVITY']==0]['CIRCULATION'].mean()

4035.958433014354

Observamos que la media de `Circulation` para la `Activity = 0` es 4035.95.

___

**Pregunta 3**

Calcula cuántos datos de entre todas las `Activity` están por encima de la media del campo `Circulation` de `Activity = 0`.

**Respuesta**

Tenemos la media de `Activity = 0`. Es la siguiente:

In [24]:
mean0 = traindf[traindf['ACTIVITY']==0]['CIRCULATION'].mean()
mean0

4035.958433014354

Comprobamos ahora cuántos valores cumplen la condición:

In [23]:
len(traindf[traindf['CIRCULATION'] > mean0])

2624

Por tanto, tenemos 2624 datos de todas las `Activity` que están por encima de la media de `Activity = 0`.

---

**Pregunta 4**

Txt

**Pregunta 5**

Txt